engine/src/cmd/planet.cpp

```diff
@@ -42,7 +44,7 @@ GamePlanet::GamePlanet() :
 
 static void SetFogMaterialColor( Mesh *thus, const GFXColor &color, const GFXColor &dcolor )
 {
-    static float emm  = XMLSupport::parse_float( vs_config->getVariable( "graphics", "atmosphere_emmissive", "1" ) );
+    static float emm  = GameConfig::GetVariable( "graphics", "atmosphere_emmissive", 1.0f);
     static float diff = XMLSupport::parse_float( vs_config->getVariable( "graphics", "atmosphere_diffuse", "1" ) );
```

Comment:
Why did your keep xml support here?

Reply:
This is not the focus of this PR. As tempting as it is to just spray code changes everywhere, I'm trying to limit my changes to a specific file or goal and resulting changes in other files. In this case, I'm really trying to refactor unit_generic to make it smaller and more encapsulated by moving specific functions to other parent classes. However, this commit includes code written a while ago and so I'm trying to get rid of a lot of WIP here too (so not really one file/goal). :)